packages/app/contexts/TokenListContext.tsx

```diff
@@ -163,23 +166,45 @@ export const TokenListProvider = ({ children }: PropsWithChildren) => {
             mergedTokenlistTokens,
             result.value.tokens
           );
+          const tokenUniqueAddresses: string[] = [];
+          const tokenListNoDuplicates: TokenFromTokenlist[] = [];
+
+          const tokenList = () => {
+            mergedTokenlistTokens.forEach((token) => {
+              if (!tokenUniqueAddresses.includes(token.address)) {
+                tokenUniqueAddresses.push(token.address);
+                tokenListNoDuplicates.push(token);
+              }
+            });
+
+            const tokenListOfCurrentChainId = tokenListNoDuplicates.filter(
+              (token: TokenFromTokenlist) => token.chainId === chainId
+            );
+
+            return tokenListOfCurrentChainId;
+          };
```

Comment:
I think we don't need to wrap it inside a function as it's only create a bit more complexity on reading this code. Passing `tokenListOfCurrentChainId` to `setTokenList` will create the same effect